### PR TITLE
fix(infra): Use proper common expression language syntax

### PR DESCRIPTION
**Pull request by jamilbk**
https://github.com/firezone/firezone/actions/runs/11713228570/job/32626046819

Language reference:

https://github.com/google/cel-spec/blob/master/doc/langdef.md#macros

### CODE REVIEW

**Review by thomaseizinger — APPROVED** [review]
LGTM

File: terraform/modules/google-cloud/apps/elixir/network.tf [hunk 1]
```diff
@@ -73,7 +73,7 @@ resource "google_compute_security_policy" "default" {
 
     match {
       expr {
-        expression = "request.path.matches(\"/sign_up\") && origin.region_code in ('RU', 'BY', 'KP', 'IR', 'SY', 'CU', 'VE', 'XC', 'XD')"
+        expression = "request.path.matches(\"/sign_up\") && ['RU', 'BY', 'KP', 'IR', 'SY', 'CU', 'VE', 'XC', 'XD'].exists(r, r == origin.region_code)"
```
Comment (bmanifold) [inline terraform/modules/google-cloud/apps/elixir/network.tf:76]: Just wondering if you read anywhere specific that the `.exists` macro works for these rules?

I found this stackoverflow which is why I ask:
https://stackoverflow.com/questions/73723639/google-cloud-armor-cannot-get-the-match-expression-to-work-for-in-operator-an

Also, in trying to input this in to the GCP web console, the `.exists` macro doesn't seem to come up in auto-complete

Comment (bmanifold) [reply terraform/modules/google-cloud/apps/elixir/network.tf:76]: In thinking about this more, I suppose worse case is that the rule doesn't apply like before, so let's just give it a go.

**Review by bmanifold — APPROVED** [review]
LGTM

**Pull request merged**